main.go

```diff
@@ -1256,36 +1256,151 @@ func getBMPPorts() (gdbPort, uartPort string, err error) {
 	}
 }
 
-func usage(command string) {
-	switch command {
-	default:
-		fmt.Fprintln(os.Stderr, "TinyGo is a Go compiler for small places.")
-		fmt.Fprintln(os.Stderr, "version:", goenv.Version())
-		fmt.Fprintf(os.Stderr, "usage: %s <command> [arguments]\n", os.Args[0])
-		fmt.Fprintln(os.Stderr, "\ncommands:")
-		fmt.Fprintln(os.Stderr, "  build:   compile packages and dependencies")
-		fmt.Fprintln(os.Stderr, "  run:     compile and run immediately")
-		fmt.Fprintln(os.Stderr, "  test:    test packages")
-		fmt.Fprintln(os.Stderr, "  flash:   compile and flash to the device")
-		fmt.Fprintln(os.Stderr, "  gdb:     run/flash and immediately enter GDB")
-		fmt.Fprintln(os.Stderr, "  lldb:    run/flash and immediately enter LLDB")
-		fmt.Fprintln(os.Stderr, "  monitor: open communication port")
-		fmt.Fprintln(os.Stderr, "  ports:   list available serial ports")
-		fmt.Fprintln(os.Stderr, "  env:     list environment variables used during build")
-		fmt.Fprintln(os.Stderr, "  list:    run go list using the TinyGo root")
-		fmt.Fprintln(os.Stderr, "  clean:   empty cache directory ("+goenv.Get("GOCACHE")+")")
-		fmt.Fprintln(os.Stderr, "  targets: list targets")
-		fmt.Fprintln(os.Stderr, "  info:    show info for specified target")
-		fmt.Fprintln(os.Stderr, "  version: show version")
-		fmt.Fprintln(os.Stderr, "  help:    print this help text")
+const (
+	usageBuild = ` Compile the given program. The output binary is specified using the -o parameter.
+The generated file type depends on the extension:
+
+	.o: 				
+					Create a relocatable object file. You can use this option if you don't want to 
+					use the TinyGo build system or want to do other custom things.
+
+	.ll: 			
+					Create textual LLVM IR, after optimization. This is mainly useful for debugging.
+
+	.bc: 			
+					Create LLVM bitcode, after optimization. This may be useful for debugging or for
+					linking into other programs using LTO.
+
+	.hex 
+					Create an Intel HEX file to flash it to a microcontroller.
```

Comment:
- limit help output to 80 chars wide (even `BigGo` does this)
- inconsistent list values (.o, .ll, and .bc end in ':' while the rest do not)
- the formatting of lists is odd, the 'definition' is indented too far

I think 'Compile the given program'  is misleading since 'program' indicates something directly runnable and the list following then contradicts that idea. In any case, BigGo says: "Build compiles the packages named by the import paths, along with their dependencies, but it does not install the results." I would think if `tinygo` is largely drop-in tooling for `BigGo` it would make most sense to follow its lead and note (only?) any different behavior?

Reply:
I think the wording is mostly pulled from the website; any fixes should be made there too.

Reply:
Indeed as @dgryski pointed out I took the existing help from the website and put it into the CLI. I agree, that the wording could use a rework.

Reply:
https://github.com/tinygo-org/tinygo-site/pull/432

Reply:
I'm not any kind of authority in this project. I just have opinions like anybody else. I'm not familiar enough with this project to approve any PR especially about documentation / usage().

I was wrong:  "go help" does exceed 80 chars, uses tabs, is tied to how wide the tab is in your terminal, and: yuck.

I'll opine that the documentation closer to the code, as here, is more accessible, likely to be up to date(?), and generally good. Having it in two places is not so good.